images/manageiq-base-worker/container-assets/manageiq_liveness_check

```diff
@@ -1,4 +1,17 @@
 #!/bin/bash
 
-[[ -s /etc/default/evm ]] && source /etc/default/evm
-ruby ${APP_ROOT}/lib/workers/bin/heartbeat_check.rb --heartbeat-file=${APP_ROOT}/tmp/worker.hb
+# We're alive if the hb_file exists and has a unix time "newer" / greater
+# than the current time.
+hb_file="${APP_ROOT}/tmp/worker.hb"
+if [[ -f ${hb_file} ]]; then
+  current_time=$(date +%s)
+  custom_timeout=$(cat ${hb_file})
```

Comment:
why call it `custom_timeout` instead of `heartbeat_time`?

Reply:
it's sort of neither.  The value inside is the custom threshold (specifically, `Time.now.to_i + heartbeat_threshold`)

Reply:
Maybe just call it `threshold`?

Reply:
Good question.  It's because we store the [current time + the worker's timeout or the default timeout of 2 minutes](https://github.com/ManageIQ/manageiq/blob/de4c23643816b171243b26c5d48dc03add943873/app/models/miq_worker/runner.rb#L316).  We're storing the "custom timeout", or not the actual last heartbeat.  It's also not the [heartbeat timeout](https://github.com/ManageIQ/manageiq/blob/de4c23643816b171243b26c5d48dc03add943873/lib/workers/miq_defaults.rb#L5) as that's a constant 2 minutes.  We have a custom value that could be the current time + that 2 minutes or a timeout from that worker's settings. 

Of note, we can't even see the worker heartbeat files from the "server" process in podified unless they're on the same filesystem, which isn't the case in pods.  The server assumes the worker's are alive or dead based on the liveness check.  If the worker row goes away, because it was killed due to not heartbeating, we won't update the worker's last heartbeat.  If it does exist, we assume it's heartbeating and set the last heartbeat to the current time.  See [here](https://github.com/ManageIQ/manageiq/pull/21079/files#diff-7872eccf60832f6d08361bdb1a0cc4ce6ea7482b5c48b17b6b145af33c78e2b0R7)

Reply:
> Maybe just call it `threshold`?

I'm not really a fan of threshold since that's also used for memory_threshold and it's not really conveying a meaning.  I think it's very much a custom timeout or maybe just a timeout.  We overloaded terms with heartbeat timeout so we can't really use that.  I'm open to other ideas.

Reply:
I think the part that is confusing is comparing `current_time` to a `timeout`.  The former feels like a particular time and the latter like a length of time (i.e. 5 minutes)

Reply:
We have agreed that `not_responsive_threshold` sounds better for the purposes of what the liveness check is doing.